ports/rp2/machine_uart.c

```diff
@@ -271,6 +339,11 @@ STATIC mp_obj_t machine_uart_make_new(const mp_obj_type_t *type, size_t n_args,
         uart_set_fifo_enabled(self->uart, true);
         gpio_set_function(self->tx, GPIO_FUNC_UART);
         gpio_set_function(self->rx, GPIO_FUNC_UART);
+        if (self->flow) {
+            gpio_set_function(self->cts, GPIO_FUNC_UART);
+            gpio_set_function(self->rts, GPIO_FUNC_UART);
+            uart_set_hw_flow(self->uart, true, true);
```

Comment:
This could be something like `uart_set_hw_flow(self->uart, self->flow & 1, self->flow & 2);`